Feature/step02

SujinHuh opened the pull request:
리뷰요청 
질문있습니다. 
검색 기능을  JpaRepository를 사용해서 Sort하는건지요...

SujinHuh committed:
1주차 과제 Refactoring
 - RestTemplate를 Bean으로 정의 @RequiredArgsConstructor으로 생성자 주입

SujinHuh committed:
2주차 과제
 - RestTemplatefmf Configuration,Bean 어노테이션 정의               - 완료
 - 영화 검색 기능에서, 유저들의 평점이 놓은 순으로 정렬
 - custom annotation properties                                  - 완료
 - local profile 실행
 - 검색 API를 호출하는 RestTemplate을  refactoring / 제네릭 사용
 - 영화 검색 평점순 Sort TDD 작성 / 오픈 api 사용 X , Mock 객체 사용

sieunkr commented:
질문 : 검색 기능을 JpaRepository를 사용해서 Sort하는건지요.. 
--> 어떤 검색 기능을 말씀하시는지요? 

과제에 있는 영화 검색에 평점순 정렬을 말씀하시는건가요? 

일단, JPA 는 전혀 사용하지 않습니다. 
오픈API 를 통해서 조회한 결과 list 를 평점이 높은 순으로 정렬해주시면 됩니다!

SujinHuh committed:
2주차 과제
 - RestTemplatefmf Configuration,Bean 어노테이션 정의                - 완료
 - 영화 검색 기능에서, 유저들의 평점이 놓은 순으로 정렬                     - 완료
 - custom annotation properties                                  - 완료
 - local profile 실행
 - 검색 API를 호출하는 RestTemplate을  refactoring / 제네릭 사용
 - 영화 검색 평점순 Sort TDD 작성 / 오픈 api 사용 X , Mock 객체 사용

sieunkr commented on src/main/java/springboot/study/MainController.java:
```diff
@@ -31,16 +31,18 @@ public MainController(BlogService blogService, MovieService movieService, Encycl
-
-        return blogService.findByQuery(query);
+    public List<BlogDto> searchBlogByQuery(){
+        return blogService.findByQuery("강남");
```
@RequestParam(name = "query") 가 다시 빠졌네요?

sieunkr commented on src/main/java/springboot/study/MainController.java:
```diff
@@ -31,16 +31,18 @@ public MainController(BlogService blogService, MovieService movieService, Encycl
 
     @GetMapping("/movie")
     public List<MovieDto> searchMovieByQuery() {
         return movieService.findByQuery("해리포터");
```
파라미터로 받도록 수정해주세요.

sieunkr commented on src/main/java/springboot/study/respons/ResponseBlog.java:
```diff
@@ -13,13 +13,12 @@
-        String bloggername;
-        String description;
-        String postdate;
+    @Data
```
Getter, Setter 어노테이션으로 변경해주세요

sieunkr commented on src/main/java/springboot/study/service/MovieService.java:
```diff
@@ -1,31 +1,42 @@
+    }
+
+    // TODO 더... 간단한 방법이 있을 것 같은데.....음... Dto 쪽으로 보내야 하나...
+    private List<MovieDto> getSortRating(List<MovieDto> list) {
```
그때 지난번에 테스트 코드 작성하는법 배우지 않으셨나요?
배우신대로 한번 작성해보시겠어요? 

배우신 방법대로 작성해보시구요

제가 다시 보고 설명 드릴게요!!

SujinHuh committed:
2주차 과제

SujinHuh commented:
생각보다 작성하기가 어렵네요..ㅜ
